docs/data/date-pickers/adapters-locale/adapters-locale.md

```diff
@@ -0,0 +1,157 @@
+  {children}
+</LocalizationProvider>;
+```
+
+{{"demo": "LocalizationDateFns.js"}}
+
+### With `luxon`
+
+For `luxon`, you have to pass the locale name to `LocalizationProvider`:
+
+```tsx
+import { AdapterLuxon } from '@mui/x-date-pickers/AdapterLuxon';
+
+<LocalizationProvider dateAdapter={AdapterLuxon} adapterLocale="de">
+  {children}
+</LocalizationProvider>;
+```
+
+{{"demo": "LocalizationLuxon.js"}}
+
+### With `moment`
+
+For `moment`, you have to import the locale and then pass its name to `LocalizationProvider`:
+
+```tsx
+import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
+import 'moment/locale/de';
+
+<LocalizationProvider dateAdapter={AdapterMoment} adapterLocale="de">
+  {children}
+</LocalizationProvider>;
+```
+
+{{"demo": "LocalizationMoment.js"}}
+
+:::warning
+Some of `moment` methods do not support scoped local.
+To have a correct localization, you will have to manually update the global local before updating it in `LocalizationProvider`.
+
+```tsx
+function App({ children }) {
+  const [locale, setLocale] = React.useState('en-us');
+
+  if (moment.locale() !== locale) {
+    moment.locale(locale);
+  }
+
+  return (
+    <LocalizationProvider dateAdapter={AdapterMoment} adapterLocale={locale}>
+      <Stack>
+        <Button onClick={() => setLocale('de')}>Switch to German</Button>
+        {children}
+      </Stack>
+    </LocalizationProvider>
+  );
+}
+```
+
+:::
+
+## 12h/24h format
+
+All the time and datetime components will automatically adjust to the locale's time setting, i.e. the 12-hour or 24-hour format.
+This can be overridden by using the `ampm` prop:
+
+{{"demo": "AmPMCustomization.js"}}
+
+## Custom formats
+
+:::warning
+The format received by the props described below depends on the date library you are using.
+Please refer to each library documentation for the full format table:
+
+- [Day.js](https://day.js.org/docs/display/format)
+- [date-fns](https://date-fns.org/docs/format)
+- [Luxon](https://moment.github.io/luxon/#/formatting?id=table-of-tokens)
+- [Moment.js](https://momentjs.com/docs/#/displaying/format/)
+
+:::
+
+### Custom field format
+
+The fields have a default format that depends on the picker being used, the views enabled and the 12h/24h format.
+
+If this default format does not suit you, you can customize it using the `format` prop:
+
+:::info
+This prop is available on all the fields and all the pickers.
+:::
+
+{{"demo": "CustomFieldFormat.js"}}
+
+### Custom toolbar format
+
+To customize the format used in the toolbar, you can use the `toolbarFormat` prop of the toolbar slot.
+
+:::info
+This prop is available on all the pickers.
+:::
+
+{{"demo": "CustomToolbarFormat.js"}}
+
+### Custom day of week format
+
+To customize day names in calendar header, you can use `dayOfWeekFormatter` which takes as an input the short name of the day provided by the date-library and returns it's formatted version.
+The default formatter only keeps the first letter and capitalises it.
```

Comment:
Thinking about it, I'm wondering if the current API is what it should be.
Passing a string feels a bit off to me because people can only do customization based on this string.

Should it be a locale key ?

```tsx
daysOfWeekShort: (utils: MuiPickersAdapter<TDate>) => string[]
```

This is slightly more verbose for very basic customization (like removing the capitalization), but people would not be bound to `getWeekdays`.
And this would use the locale which is more coherent to me here.

```tsx
// Default one
daysOfWeekShort: (utils) => utils.getWeekDays().map(day => day.charAt(0).toUpperCase());

// Without capitalization
daysOfWeekShort: (utils) => utils.getWeekDays().map(day => day.harAt(0));

// Or even hardcode if the locale is not working great with `getWeekDays`
daysOfWeekShort: () => SOME_CUSTOM_ARRAY;
```

Reply:
I would agree that the current implementation is pretty limiting due to the fact that we are working with `pre-formatted` data.
Your suggestion sounds more useful, but it would still not solve [issues like this one](https://github.com/mui/mui-x/issues/6461).
I'm just not sure how much use would it get. 🤔 

And in any case—I'd reckon such a prop should at best be an addition with MAYBE deprecating `dayOfWeekFormatter`.

Reply:
I don't think it's something urgent, and I would agree to deprecate so the breaking change would be for v7 anyway.

Bur for now, I don't know if we should advertise about the adapter `getWeekDays` method.

Reply:
Case settled. 👌 
I just mentioned this, because I'm biased by the needed investigation on the mentioned issue. 🙈

Reply:
Let's say that, if advertising about the `getWeekDays` adapter method is a way to say to people "you can provide a customized adapter with a custom `geteekDays` method in case `dayOfWeekFormatter` do not work for you"

Then I think it's a bad solution, and the right solution would be, as you said, to let people work with non formatted data, without touching the adapter.